lib/es-models/item.js

```diff
@@ -0,0 +1,372 @@
+  }
+
+  _accessMessageCode () {
+    let accessMessageCode = null
+    const item = this.item
+    if (item.isPartnerRecord()) {
+      let message = item.varField('876', ['h'])
+      if (message && message.length > 0) {
+        message = message.pop()
+
+        // If 876 $h is 'IN LIBRARY USE' or [blank]:
+        if (message.toLowerCase() === 'in library use' || message.toLowerCase() === '') {
+          accessMessageCode = '1'
+
+          // If 876 $h is SUPERVISED USE:
+        } else if (message.toLowerCase() === 'supervised use') {
+          accessMessageCode = 'u'
+
+          // If 876 $h is set to anything else, try using fixed "OPAC Message"
+        } else if (item.fixed('OPAC Message')) {
+          accessMessageCode = item.fixed('OPAC Message')
+        }
+      }
+
+      // If we didn't find the partner access message in 876 $h, default to In Library Use:
+      if (!accessMessageCode) {
+        accessMessageCode = '1'
+      }
+
+      // Otherwise it's ours; look for OPAC Message:
+    } else if (item.fixed('OPAC Message')) {
+      accessMessageCode = item.fixed('OPAC Message')
+    }
+
+    return accessMessageCode
+  }
+
+  // requires utils.coreObjectsMappingByCode
+  accessMessage () {
+    const accessMessageCode = this._accessMessageCode()
+
+    if (accessMessageCode) {
+      // Look up code in accessmessages to make sure it's valid:
+      const mapping = coreObjectsMappingByCode(accessMessageMapping, accessMessageCode)
+      if (mapping) {
+        const accessmessageId = mapping.id.split(':').pop()
+        return [{ id: `accessMessage:${accessmessageId}`, label: mapping.label }]
+      } else logger.error('unmapped opac message? ' + accessMessageCode)
+    }
+
+    return null
+  }
+
+  // check pack
+  accessMessage_packed () {
+    return pack(this.accessMessage())
+  }
+
+  catalogItemType () {
+    const itemType = this.item.getItemType()
+
+    if (itemType) {
+      const catalogItemType = coreObjectsMappingByCode(catalogItemTypeMapping, String(itemType))
+      if (catalogItemType) {
+        const catalogItemTypeId = catalogItemType.id.split(':').pop()
+        const catalogItemTypeEntity = { id: `catalogItemType:${catalogItemTypeId}`, label: catalogItemType.label }
+        return [catalogItemTypeEntity]
+      }
+    }
+
+    return null
+  }
+
+  catalogItemType_packed () {
+    return pack(this.catalogItemType())
+  }
+
+  enumerationChronology () {
+    if (this.item.fieldTag('v') && this.item.fieldTag('v').length) {
+      return [this.item.fieldTag('v')[0].value]
+    }
+
+    return null
+  }
+
+  _location () {
+    if (this.location) return this.location
+    let location = null
+    const item = this.item
+    if (item.location) {
+      location = item.location.code
+    } else {
+      const altLocation = item.varField('852', ['b'])
+      if (altLocation && altLocation.length > 0) location = altLocation
+    }
+
+    if (location && sierraLocationMapping[location]) {
```

Comment:
this conditional might be clearer if it was:
```
if (location) {
   if (sierraLocationMapping[location]){}
   else 
```

Reply:
I dislike nesting conditionals if I can avoid it, do you think that would be much clearer? @nonword any style preferences regarding nesting conditionals vs not?

Reply:
ill defer to that preference!

Reply:
I think both nested and compound conditionals kind of have the same cognitive complexity problem. Given two+ variables, it's a challenge to keep track of what conditions will lead one where. I prefer to address this by ruling out the simple (i.e. error) conditions first, like:
```
...
// No location [code] extracted? Nothing to build:
if (!location) return null
// Location code isn't recognized? Nothing to build:
if (!sierraLocationMapping[location]) {
  if (item.isNyplRecord()) logger.warn('Location id not recognized: ', location)
  return null
}
// At this point we know the location code is valid, so build the entity:
const holdingLocationId = `loc:${location}`
const holdingLocationLabel = sierraLocationMapping[location].label
this.location = { id: holdingLocationId, label: holdingLocationLabel, rawId: location }
return this.location
```
But I'm also okay with this specific block as-is because it's not all that complex